src/Pipeline.h

```diff
@@ -443,23 +443,32 @@ void init_arg_types(std::vector<ScalarOrBufferT> &arg_types) {
 
 }
 
+struct ExternCFunction {
+    void *address{nullptr};
+    ExternSignature signature;
+
+    ExternCFunction() = default;
+
+    template <typename RT, typename... Args>
+    ExternCFunction(RT (*f)(Args... args)) {
+        address = (void *)f;
+        signature.is_void_return = voidable_halide_type<RT>(signature.ret_type);
+        init_arg_types<Args...>(signature.arg_types);
+    }
+};
+
 struct JITExtern {
-    // assert pipeline.defined() == (c_function == nullptr) -- strictly one or the other
+    // assert pipeline.defined() == (extern_c_function.address == nullptr) -- strictly one or the other
```

Comment:
If this constraint is important, perhaps this should be a class where these fields are private, and the constructors only set one or the other. I prefer to avoid constraints like this that are only communicated via comments.